src/utils/eth_transaction.cairo

```diff
@@ -0,0 +1,264 @@
+        // 1 byte for v
+        // 33 bytes for r = 1 byte for len (=32) + 32 bytes for r word
+        // 33 bytes for s = 1 byte for len (=32) + 32 bytes for s word
+        local signature_len = 1 + 1 + 32 + 1 + 32;
+        local signature_start_index = tx_type + 7;
+        local chain_id_idx = 0;
+        // 1. extract v, r, s
+        let (chain_id) = Helpers.bytes_to_felt(
+            sub_items[chain_id_idx].data_len, sub_items[chain_id_idx].data, 0
+        );
+        assert chain_id = Constants.CHAIN_ID;
+        let (v) = Helpers.bytes_to_felt(
+            sub_items[signature_start_index].data_len, sub_items[signature_start_index].data, 0
+        );
+        let r = Helpers.bytes32_to_uint256(sub_items[signature_start_index + 1].data);
+        let s = Helpers.bytes32_to_uint256(sub_items[signature_start_index + 2].data);
+
+        let (local signed_data: felt*) = alloc();
+        assert [signed_data] = tx_type;
+        let (rlp_len) = RLP.encode_list(
+            [items].data_len - signature_len, [items].data, signed_data + 1
+        );
+
+        let (local words: felt*) = alloc();
+        let (keccak_ptr: felt*) = alloc();
+        let keccak_ptr_start = keccak_ptr;
+        with keccak_ptr {
+            // From keccak/keccak_bigend doc:
+            // > To use this function, split the input into words of 64 bits (little endian).
+            // > Same as keccak, but outputs the hash in big endian representation.
+            // > Note that the input is still treated as little endian.
+            Helpers.bytes_to_bytes8_little_endian(
+                bytes_len=rlp_len + 1,
+                bytes=signed_data,
+                index=0,
+                size=rlp_len + 1,
+                bytes8=0,
+                bytes8_shift=0,
+                dest=words,
+                dest_index=0,
+            );
+            let (tx_hash) = keccak_bigend(inputs=words, n_bytes=rlp_len + 1);
+        }
+        finalize_keccak(keccak_ptr_start, keccak_ptr);
+
+        let gas_limit_idx = tx_type + 2;
+        let (gas_limit) = Helpers.bytes_to_felt(
+            sub_items[gas_limit_idx].data_len, sub_items[gas_limit_idx].data, 0
+        );
+        let (destination) = Helpers.bytes_to_felt(
+            sub_items[gas_limit_idx + 1].data_len, sub_items[gas_limit_idx + 1].data, 0
+        );
+        let (amount) = Helpers.bytes_to_felt(
+            sub_items[gas_limit_idx + 2].data_len, sub_items[gas_limit_idx + 2].data, 0
+        );
+        let payload_len = sub_items[gas_limit_idx + 3].data_len;
+        let payload: felt* = sub_items[gas_limit_idx + 3].data;
+        return (gas_limit, destination, amount, payload_len, payload, tx_hash, v, r, s);
+    }
+
+    func is_legacy_tx{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        bitwise_ptr: BitwiseBuiltin*,
+        range_check_ptr,
+    }(tx_data: felt*) -> felt {
+        tempvar type = [tx_data];
+        if ((type - 2) * (type - 1) == 0) {
```

Comment:
I guess this is valid for now
We can differentiate legacy txs from EIP2718 ones using: https://eips.ethereum.org/EIPS/eip-2718#transactiontype-only-goes-up-to-0x7f

> For the forseable future, 0x7f is plenty and it leaves open a number of options for extending the range such as using the high bit as a continuation bit. This also prevents us from colliding with legacy transaction types, which always start with a byte >= 0xc0.

Reply:
If type > 0xc0 then it is legacy